packages/qwik-city/src/middleware/request-handler/resolve-request-handlers.ts

```diff
@@ -355,23 +360,24 @@ async function pureServerFunction(ev: RequestEvent) {
 
 function fixTrailingSlash(ev: RequestEvent) {
   const trailingSlash = getRequestTrailingSlash(ev);
-  const { basePathname, pathname, url, sharedMap } = ev;
+  const { basePathname, originalUrl, sharedMap } = ev;
+  const { pathname, search } = originalUrl;
   const isQData = sharedMap.has(IsQData);
   if (!isQData && pathname !== basePathname && !pathname.endsWith('.html')) {
     // only check for slash redirect on pages
     if (trailingSlash) {
       // must have a trailing slash
       if (!pathname.endsWith('/')) {
         // add slash to existing pathname
-        throw ev.redirect(HttpStatus.MovedPermanently, pathname + '/' + url.search);
+        throw ev.redirect(HttpStatus.MovedPermanently, pathname + '/' + search);
```

Comment:
actually, shouldn't the search come from the current URL? It might have changed, no?

Reply:
Yes, good catch, will fix it

Reply:
It would actually be equal(by design), as I said before, maybe you missed it the thread has become very long.
The search param is loosely obtained, unlike path params. For DX reasons I think the search params should always stay as the search params from the URL. as opposed to the path params which are always obtained in the same way from a framework specific api. There is no use case for rewriting search params(that I know of), and it feels much more confusing to support such a thing.

Reply:
ok fair.